### PR TITLE
[POC] Support for comparing multivalue numeric fields

### PR DESCRIPTION
**Pull request by dantleech**
This is just a POC for now.

### CODE REVIEW

File: src/Jackalope/Transport/DoctrineDBAL/Query/QOMWalker.php [hunk 1]
```diff
@@ -627,10 +627,26 @@ public function walkNumComparisonConstraint(QOM\PropertyValueInterface $property
...
         $property = $propertyOperand->getPropertyName();
 
-        return
-            $this->sqlXpathExtractNumValue($alias, $property) . " " .
-            $operator . " " .
-            $literalOperand->getLiteralValue();
+        if ($this->platform instanceof MySqlPlatform) {
+            switch ($operator) {
+                case '=':
+                case '!=':
```
Comment (dantleech) [inline src/Jackalope/Transport/DoctrineDBAL/Query/QOMWalker.php]: Ignore this switch case..

**Comment by dantleech** [comment]
@dbu @lsmith77 what do you think to this approach?

**Comment by lsmith77** [comment]
if it fixes the issue, then its fine to me. however I wonder if we have the issue with SQLite/PostgreSQL as well? I guess a test case makes sense ..

**Comment by dantleech** [comment]
I imagine we will, lets add a test case and see how it goes.

**Comment by dbu** [comment]
seems to only work for postgres. and we seem to have some issue with reloading fixtures.

**Comment by dantleech** [comment]
Thats odd, I am pretty sure I developed it in MySQL. Will have another look soon regardless.

**Comment by dbu** [comment]
i added this to the 1.2 milestone, as lukas says this is important for sulu. is it?

**Comment by dantleech** [comment]
yeah, it would be very useful /cc @watcherjohannes

**Comment by dantleech** [comment]
ok this should pass now, although I suspect we still have a flaky test lurking in master. (`/container/idExample` not found).

This PR also makes me think that we should handle the query building with events, or at least provide a more structured way of specializing queries for different drivers. Maybe something to think about.

**Comment by dbu** [comment]
the flaky test is noted in #239

**Comment by dbu** [comment]
mysql reliably fails. is this related to what you are doing or not?

**Comment by dantleech** [comment]
yeah sorry -- just found out I was only testing locally on sqlite. debugging MySQL now.

**Comment by dantleech** [comment]
ok, builds now passing apart from the flaky sqlite.

File: src/Jackalope/Transport/DoctrineDBAL/Query/QOMWalker.php [hunk 2]
```diff
@@ -627,10 +627,28 @@ public function walkNumComparisonConstraint(QOM\PropertyValueInterface $property
         $alias = $this->getTableAlias($propertyOperand->getSelectorName() . '.' . $propertyOperand->getPropertyName());
         $property = $propertyOperand->getPropertyName();
 
-        return
-            $this->sqlXpathExtractNumValue($alias, $property) . " " .
-            $operator . " " .
-            $literalOperand->getLiteralValue();
+
+        if ($this->platform instanceof MySqlPlatform) {
```
Comment (dbu) [inline src/Jackalope/Transport/DoctrineDBAL/Query/QOMWalker.php]: to make this more readable, could we do?

```
if ($this->platform instanceof MySqlPlatform && in_array($operator, array('=', '!='))) {
    return sprintf(...
}
```

**Comment by dbu** [comment]
great. apart from the readability i think this is good now. i really would merge the if and the switch into one, about the assignment vs direct return i'd prefer the direct return but don't mind too much.

can you then squash the commits please?

**Comment by dantleech** [comment]
Restructured the code. Also found out that `!=` is not valid JCR-SQL2. So have just made it act only on equality for now.

**Comment by dantleech** [comment]
and squashed.

File: src/Jackalope/Transport/DoctrineDBAL/Query/QOMWalker.php [hunk 3]
```diff
@@ -627,10 +627,22 @@ public function walkNumComparisonConstraint(QOM\PropertyValueInterface $property
         $alias = $this->getTableAlias($propertyOperand->getSelectorName() . '.' . $propertyOperand->getPropertyName());
         $property = $propertyOperand->getPropertyName();
 
-        return
-            $this->sqlXpathExtractNumValue($alias, $property) . " " .
-            $operator . " " .
-            $literalOperand->getLiteralValue();
+
+        if ($this->platform instanceof MySqlPlatform && $operator == '=') {
```
Comment (lsmith77) [inline src/Jackalope/Transport/DoctrineDBAL/Query/QOMWalker.php]: ``if ($this->platform instanceof MySqlPlatform && $operator == '=') {``

=>

``if ($this->platform instanceof MySqlPlatform && '=' === $operator) {``

Comment (dantleech) [inline src/Jackalope/Transport/DoctrineDBAL/Query/QOMWalker.php]: Does `===` offer any advantage here? Changed anyway.

Comment (lsmith77) [inline src/Jackalope/Transport/DoctrineDBAL/Query/QOMWalker.php]: its slightly faster and of course its type safe. I just find it a good practice to "default" to ``===``

**Comment by dantleech** [comment]
good to merge?

**Comment by dbu** [comment]
great, thanks a lot!

**Pull request merged**